### PR TITLE
feat(mnq): add sns

### PR DESCRIPTION
**Pull request by Codelax**

### CODE REVIEW

File: docs/resources/mnq_sns_topic_subscription.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```hcl
+// For default project in default region
+resource "scaleway_mnq_sns" "main" {}
+
+resource scaleway_mnq_sns_credentials main {
+  project_id = scaleway_mnq_sns.main.project_id
+  permissions {
+    can_manage = true
+    can_publish = true
+	can_receive = true
```
Comment (remyleone) [inline docs/resources/mnq_sns_topic_subscription.md:25]:
```suggestion
  can_receive = true
```

File: docs/resources/mnq_sns_topic_subscription.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+```hcl
+// For default project in default region
+resource "scaleway_mnq_sns" "main" {}
+
+resource scaleway_mnq_sns_credentials main {
+  project_id = scaleway_mnq_sns.main.project_id
+  permissions {
+    can_manage = true
+    can_publish = true
+  can_receive = true
```
Comment (remyleone) [inline docs/resources/mnq_sns_topic_subscription.md:25]: indentatino issue here

File: scaleway/helpers_mnq.go [hunk 3]
```diff
@@ -48,5 +53,226 @@ func mnqSQSAPIWithRegionAndID(m interface{}, regionalID string) (*mnq.SqsAPI, sc
...
+}
+
+func (a ARN) String() string {
+	if a.ExtraResourceID == "" {
+		return fmt.Sprintf("arn:scw:%s:%s:project-%s:%s", a.Subject, a.Region, a.ProjectID, a.ResourceName)
+	}
+	return fmt.Sprintf("arn:scw:%s:%s:project-%s:%s:%s", a.Subject, a.Region, a.ProjectID, a.ResourceName, a.ExtraResourceID)
+}
+
+// decomposeARN decompose an arn with a potential extra-resource-id
```
Comment (remyleone) [inline scaleway/helpers_mnq.go:117]:
```suggestion
// decomposeARN decomposes an arn with a potential extra-resource-id
```

File: docs/resources/mnq_sns_topic_subscription.md (hunk 2 above)
Comment (remyleone) [inline docs/resources/mnq_sns_topic_subscription.md:25]: indentation error

File: scaleway/resource_mnq_sns_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+				l.Debugf("sweeper: error (%s)", err)
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+func TestAccScalewayMNQSNS_Basic(t *testing.T) {
```
Comment (remyleone) [inline scaleway/resource_mnq_sns_test.go:52]: Can it be deactivated? Could you add a test step for that?

File: scaleway/resource_mnq_sns_topic.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+			},
+			"fifo_topic": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Optional:    true,
+				Description: "Whether the topic is a FIFO topic. If true, the topic name must end with .fifo",
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Computed: true,
```
Comment (remyleone) [inline scaleway/resource_mnq_sns_topic.go:73]: Could you add a description here?

File: scaleway/resource_mnq_sns_topic.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
```
Comment (remyleone) [inline scaleway/resource_mnq_sns_topic.go:77]: Could you add a description here?

File: scaleway/resource_mnq_sns_topic_subscription.go [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+			"redrive_policy": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Optional:    true,
+				Description: "JSON Redrive policy",
+				ForceNew:    true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
```
Comment (remyleone) [inline scaleway/resource_mnq_sns_topic_subscription.go:79]: Could you add a description here? In particular with the format that will be returned

File: scaleway/resource_mnq_sns_test.go (hunk 4 above)
Comment (Codelax) [reply scaleway/resource_mnq_sns_test.go:52]: Deactivation is the destroy step, it is checked by the CheckDestroy function

**Review by remyleone — APPROVED** [review]

**Pull request merged**